### DISCUSSION TITLE
Arduino 2?

### DISCUSSION THREAD

**Discussion by yesnoj** [Bodmer/TFT_eSPI#2679] [Q&A - General]
Hi guys, 
sorry for the noob question but, i've worked a lot on a project with my Arduino 2 and a ILI9341 Mcufriend_kvb 2.4" display.
For some reason i need to switch to TFT_eSPI over the Adafruit_GFX, can i do this switch or i need to find a different combo of Arduino and display?
Thanks a lot in advance!

**Comment by Bodmer** [comment]
The Arduino Due could be used with TFT_eSPI, however it does not have the processing power or RAM size to make use of the more advanced features. For the Due it would be better to stick with Adafruit_GFX.

**Reply by yesnoj** [reply]
> The Arduino Due could be used with TFT_eSPI, however it does not have the processing power or RAM size to make use of the more advanced features. For the Due it would be better to stick with Adafruit_GFX.

Ciao Bodmer!
I didn't expect an answer from you, glad to read your answer!
I had a conversation with Paul Conti, the developer of the GUISlice builder software, i've switched my entire project to use that builder, but at the end of my coding, i had an issue with the bargraphs (https://github.com/ImpulseAdventure/GUIslice/discussions/513 here the issue explained), and he told me that to fix that issue i need to switch to ESP32 board, because his fix works on that platform....so i need to switch again ...
Unfortunally the ESP32 WROOM Devkit V1 that i've here, doesn't have the 20pins that i need to manage all my relays and sensors/motors...so i'm actually stuck on this step...
I've switched to GUISlice because if in future i want to use a bigger screen (2.4" now) i can easly move components via pc software instead of use coordinates by hands.
If you have a suggestion this will be so much appreciated...i need to finish this project asap, i'm spending money on devices, pumps, motors and solenoid valves, but if i don't have a decent display (4"-5" will be perfect) with a board that has all the pins ...i can't go ahead :/

F.

**Reply by Bodmer** [reply]
If you want to plug in an UNO style board then an STM32 Nucleo board may be a good option.
https://www.st.com/en/ecosystems/stm32-nucleo.html

Alternatively if you need WiFi then an ESP32 S3 board has circa 25 GPIO available.

**Reply by Bodmer** [reply]
For Nucleo boards I suggest either a STM32F446 or STM32F767 based board depending on the processing power needed.

For ESP32 S3 this board might have enough GPIO for your needs.
https://www.mouser.co.uk/ProductDetail/Espressif-Systems/ESP32-S3-DevKitC-1-N8?qs=Wj%2FVkw3K%252BMCTHFMHLvA1pQ%3D%3D&mgh=1&vip=1&gclid=Cj0KCQjw4NujBhC5ARIsAF4Iv6eayVNhlGdHQJiY1aPOkZVjFB6ZkZI4C3s5sJM3aG5rwWDR5DB1J94aAkr3EALw_wcB

**Reply by yesnoj** [reply]
Thanks Bodmer for your hints!The Nucleo probably is the best option. the ESP32 S3 has many GPIO but if i remember well , about 8 are used to connect the tft screen.
What about a decent 4" or 5" screen to connect to the Nucleo?The price isn't an issue, the only requirement is the compatibilty with the ESP32 libraries :/

**Reply by yesnoj** [reply]
I've ordered the ESP32-S3-DevKitC-1-N8, hope arrives asap, and also the screen https://it.aliexpress.com/item/1005004152737300.html?spm=a2g0o.order_list.order_list_main.10.10b93696f1B815&gatewayAdapt=glo2ita the 3.95" with touch...i can't wait to receive it and start again the project 
The Nucleo wasn't so easy to work with...i've also ordered the expansion for the GPIO 
in case the pins are not enough on the ESP32 https://it.aliexpress.com/item/32879295871.html?spm=a2g0o.order_list.order_list_main.5.3be836961cF8To&gatewayAdapt=glo2ita

**Comment by yesnoj** [comment]
Ciao Bodmer!
Hoper you are doing well!
Just arrived the hardware to continue my project, but i'm really struggling on how connect the ESP32 with the ILI9488 display

The ESP32 i've received is the following : https://docs.espressif.com/projects/esp-idf/en/v4.4.3/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html

The display i'm trying to connect is : https://it.aliexpress.com/item/1005003033841623.html , it has XPT2046 chip on board.

Can you help me with the first setup? I've tryied different stuff, but i can't get the touch working.
I can see the images, but propably my pins are not well connected.
This is how are now linked :

ESP32 → DISPLAY
3.3 → VDO
G → GND
4 → CS
5 → RST
6 → D/C
11 → SDI & TDI
12 → SCK & TCK
3.3 → BL
15 → TCS
16 → TDO
14 → PEN

I neither can't find any calibration.ino file, that suppose is needed.
With the Arduino 2 and the ILI9341 was so simple :(
Thanks in advance!
F.

**Reply by yesnoj** [reply]
finally i’ve found how to connect the board with the display, here is the list of pins :

ESP32 S3 → DISPLAY

3.3→ VDO
G → GND
10 → CS
6 → RST
7 → D/C
11 → SDI (MOSI) & TDI
12 → SCK & TCK
3.3 → BL
Not Used → SDO (MISO)
16 → TCS
13 → TDO
Not Used → PEN
I’ve used the Setup21_ILI9488.h config file from TFT_eSPI library , and defined in this way :

```
// See SetupX_Template.h for all options available
#define USER_SETUP_ID 21

#define ILI9488_DRIVER

//#define TFT_INVERSION_OFF

//WORKING ON ESP32
//#define TFT_MISO 19 // (leave TFT SDO disconnected if other SPI devices share MISO)
//#define TFT_MOSI 23
//#define TFT_SCLK 18
//#define TFT_CS   15  // Chip select control pin
//#define TFT_DC   2  // Data Command control pin
//#define TFT_RST  4  // Reset pin (could connect to RST pin)
//#define TOUCH_CS 22

//WORKING ON ESP32S3
#define TFT_CS   10 //     10 or 34 (FSPI CS0) 
#define TFT_MOSI 11 //     11 or 35 (FSPI D)
#define TFT_SCLK 12 //     12 or 36 (FSPI CLK)
#define TFT_MISO 13 //     13 or 37 (FSPI Q)
#define TFT_DC    7
#define TFT_RST   6
#define TOUCH_CS 16 // Optional for touch screen

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT


// #define SPI_FREQUENCY  20000000
#define SPI_FREQUENCY  27000000
// #define SPI_FREQUENCY  40000000
// #define SPI_FREQUENCY  80000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  16000000

#define SPI_TOUCH_FREQUENCY  2500000
```
I’ve enabled the

`#define ILI9488_DRIVER`

in the file User_Setup.h

and enabled

`#include <User_Setups/Setup21_ILI9488.h>`

in the file User_Setup_Select.h

Actually i’ve tested the On_Off_Button.ino, and the calibration was done without issue.
However i belive that my screen is not good…the display is always flickering if i barely touch it…

I'm trying now to have the SD card working...
I've found an example where the pins used are the following for ESP32 devkit board :  

5→ CS 
18 → CLK
19 → MISO
23 → MOSI

how can i wire on ESP32 S3 instead? 
I've changed the in : 

1→ CS 
12 → CLK
13 → MISO
11 → MOSI

But i've got : [735590][E][vfs_api.cpp:24] open(): File system is not mounted 
in the serial monitor....  :(

**Reply by riacob** [reply]
I believe reading [this](https://github.com/Bodmer/TFT_eSPI/discussions/898) will solve your mounting issue.
